File: examples/modulith-clinical-tool-jpa/src/main/java/io/zenwave360/example/clinicaltool/modules/surveys/core/implementation/SurveysBackofficeServiceImpl.java
package io.zenwave360.example.clinicaltool.modules.surveys.core.implementation;

import io.zenwave360.example.clinicaltool.modules.surveys.core.domain.*;
import io.zenwave360.example.clinicaltool.modules.surveys.core.inbound.*;
import io.zenwave360.example.clinicaltool.modules.surveys.core.inbound.dtos.*;
import io.zenwave360.example.clinicaltool.modules.surveys.core.implementation.mappers.*;
import io.zenwave360.example.clinicaltool.modules.surveys.core.outbound.jpa.*;

import java.math.*;
import java.time.*;
import java.util.*;
import org.mapstruct.factory.Mappers;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

/**
 * Service Implementation for managing [Survey, Question].
 */
@Service
@Transactional(readOnly = true)
@lombok.AllArgsConstructor
public class SurveysBackofficeServiceImpl implements SurveysBackofficeService {

    private final Logger log = LoggerFactory.getLogger(getClass());

    private final SurveysBackofficeServiceMapper surveysBackofficeServiceMapper = SurveysBackofficeServiceMapper.INSTANCE;


    private final SurveyRepository surveyRepository;

    private final QuestionRepository questionRepository;






    public List<Survey> listSurveys()
 {
    log.debug("Request listSurveys");

    var surveys = surveyRepository.findAll();
    return surveys;


}


    public Optional<Survey> getSurvey(Long id)
 {
    log.debug("[CRUD] Request to get Survey : {}", id);
    var survey = surveyRepository.findById(id);
    return survey;


}

    @Transactional

    public Survey createSurvey(Survey input)
 {
    log.debug("[CRUD] Request to save Survey: {}", input);
    var survey = surveysBackofficeServiceMapper.update(new Survey(), input);
    survey = surveyRepository.save(survey);
    // TODO: may need to reload the entity to fetch relationships 'mapped by id'
    return survey;


}

    @Transactional

    public Optional<Survey> updateSurvey(Long id, Survey input)
 {
    log.debug("Request updateSurvey: {} {}", id, input);

    var survey = surveyRepository.findById(id).map(existingSurvey -> {
        return surveysBackofficeServiceMapper.update(existingSurvey, input);
    })
    .map(surveyRepository::save)
    ;
    return survey;


}

    @Transactional

    public void deleteSurvey(Long id)
 {
    log.debug("Request deleteSurvey: {}", id);

    log.debug("Request to delete Survey : {}", id);
    surveyRepository.deleteById(id);


}


    public List<Question> listQuestions()
 {
    log.debug("Request listQuestions");

    var questions = questionRepository.findAll();
    return questions;


}


    public Optional<Question> getQuestion(Long id)
 {
    log.debug("[CRUD] Request to get Question : {}", id);
    var question = questionRepository.findById(id);
    return question;


}

    @Transactional

    public Question createQuestion(Question input)
 {
    log.debug("[CRUD] Request to save Question: {}", input);
    var question = surveysBackofficeServiceMapper.update(new Question(), input);
    question = questionRepository.save(question);
    // TODO: may need to reload the entity to fetch relationships 'mapped by id'
    return question;


}

    @Transactional

    public Optional<Question> updateQuestion(Long id, Question input)
 {
    log.debug("Request updateQuestion: {} {}", id, input);

    var question = questionRepository.findById(id).map(existingQuestion -> {
        return surveysBackofficeServiceMapper.update(existingQuestion, input);
    })
    .map(questionRepository::save)
    ;
    return question;


}

    @Transactional

    public void deleteQuestion(Long id)
 {
    log.debug("Request deleteQuestion: {}", id);

    log.debug("Request to delete Question : {}", id);
    questionRepository.deleteById(id);


}



}

File: examples/modulith-klinical-tool-jpa/src/main/kotlin/io/zenwave360/example/clinicaltool/modules/surveys/service/impl/SurveysBackofficeServiceImpl.kt
package io.zenwave360.example.clinicaltool.modules.surveys.service.impl

import io.zenwave360.example.clinicaltool.modules.surveys.domain.*
import io.zenwave360.example.clinicaltool.modules.surveys.repository.jpa.*
import io.zenwave360.example.clinicaltool.modules.surveys.service.*
import io.zenwave360.example.clinicaltool.modules.surveys.service.dtos.*
import io.zenwave360.example.clinicaltool.modules.surveys.service.impl.mappers.*
import java.math.*
import java.time.*
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.data.repository.findByIdOrNull
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

/** Service Implementation for managing [Survey, Question]. */
@Service
@Transactional(readOnly = true)
open class SurveysBackofficeServiceImpl(
    private val surveyRepository: SurveyRepository,
    private val questionRepository: QuestionRepository,
) : SurveysBackofficeService {

    private val log: Logger = LoggerFactory.getLogger(javaClass)

    private val surveysBackofficeServiceMapper: SurveysBackofficeServiceMapper = SurveysBackofficeServiceMapper.INSTANCE

    override fun listSurveys(): List<Survey> {
        log.debug("Request listSurveys")

        val surveys = surveyRepository.findAll()
        return surveys
    }

    override fun getSurvey(id: Long): Survey? {
        log.debug("[CRUD] Request to get Survey : {}", id)
        val survey = surveyRepository.findByIdOrNull(id)
        return survey
    }

    @Transactional
    override fun createSurvey(input: Survey): Survey {
        log.debug("[CRUD] Request to save Survey: {}", input)
        var survey = surveysBackofficeServiceMapper.update(Survey(), input)
        survey = surveyRepository.save(survey)
        // TODO: may need to reload the entity to fetch relationships 'mapped by id'
        return survey
    }

    @Transactional
    override fun updateSurvey(id: Long, input: Survey): Survey? {
        log.debug("Request updateSurvey: {} {}", id, input)

        return surveyRepository
            .findByIdOrNull(id)
            ?.let { existingSurvey -> surveysBackofficeServiceMapper.update(existingSurvey, input) }
            ?.let { surveyRepository.save(it) }
    }

    @Transactional
    override fun deleteSurvey(id: Long): Unit {
        log.debug("Request deleteSurvey: {}", id)

        log.debug("Request to delete Survey : {}", id)
        surveyRepository.deleteById(id)
    }

    override fun listQuestions(): List<Question> {
        log.debug("Request listQuestions")

        val questions = questionRepository.findAll()
        return questions
    }

    override fun getQuestion(id: Long): Question? {
        log.debug("[CRUD] Request to get Question : {}", id)
        val question = questionRepository.findByIdOrNull(id)
        return question
    }

    @Transactional
    override fun createQuestion(input: Question): Question {
        log.debug("[CRUD] Request to save Question: {}", input)
        var question = surveysBackofficeServiceMapper.update(Question(), input)
        question = questionRepository.save(question)
        // TODO: may need to reload the entity to fetch relationships 'mapped by id'
        return question
    }

    @Transactional
    override fun updateQuestion(id: Long, input: Question): Question? {
        log.debug("Request updateQuestion: {} {}", id, input)

        return questionRepository
            .findByIdOrNull(id)
            ?.let { existingQuestion -> surveysBackofficeServiceMapper.update(existingQuestion, input) }
            ?.let { questionRepository.save(it) }
    }

    @Transactional
    override fun deleteQuestion(id: Long): Unit {
        log.debug("Request deleteQuestion: {}", id)

        log.debug("Request to delete Question : {}", id)
        questionRepository.deleteById(id)
    }
}
